bpaotu/bpaotu/views.py

```diff
@@ -555,6 +584,35 @@ def yield_csv_function():
     return response
 
 
+def _otu_endpoint_verification(data):
+    hash_portion = data.split('||')[0]
+    data_portion = data.split('||')[1]
+
+    json_data = json.loads(data_portion)
```

Comment:
verify the hash before you do anything with the untrusted content. so move that `digest == hash_portion` check up above the json decoding - we don't want to start processing user provided data until we've verified the signature

Reply:
Fixed.